xbmc/platform/posix/utils/SharedMemory.cpp

```diff
@@ -15,6 +15,9 @@
 #if defined(HAVE_LINUX_MEMFD)
 #include <linux/memfd.h>
 #include <sys/syscall.h>
+#ifndef MFD_NOEXEC_SEAL
```

Comment:
This should likely be based on a kernel version.

Reply:
That would be kernel 6.3, but I assume that some "embedded kernels" would backport it?

Reply:
I created a small test script to test the results (and subsequently updated the PR to handle both kernels with and without support for MFD_NOEXEC_SEAL - as the kernel that Kodi is compiled under doesn’t necessarily dictate the kernel Kodi runs under)

test code
```
#include <fcntl.h>
#include <sys/mman.h>
#include <unistd.h>

#include <linux/memfd.h>
#include <stdio.h>
#include <sys/syscall.h>
#ifndef MFD_NOEXEC_SEAL
#define MFD_NOEXEC_SEAL 0x0008U
#endif

int main()
{
  int fd = syscall(SYS_memfd_create, "kodi", MFD_CLOEXEC | MFD_ALLOW_SEALING | MFD_NOEXEC_SEAL);
  printf ("with MFD_NOEXEC_SEAL ret = %d\n", fd);
  if (fd < 0)
  {
    fd = syscall(SYS_memfd_create, "kodi", MFD_CLOEXEC | MFD_ALLOW_SEALING);
    printf ("without MFD_NOEXEC_SEAL ret = %d\n", fd);
  }
  return fd;
}
```

results:
```
# uname -r;./test
6.5.0-rc5
with MFD_NOEXEC_SEAL ret = 3

# uname -r;./test
6.1.38
with MFD_NOEXEC_SEAL ret = -1
without MFD_NOEXEC_SEAL ret = 3
```

Reply:
Yeah. I would still suggest fixing the cheroot env. If you are in a proper yocto environment make sure linux-libc-headers are properly created from your target kernel and not from native.

While in general I agree, if the define can be checked, that will include benefits for more users including those with backported kernels.

Reply:
> Yeah. I would still suggest fixing the chroot env. If you are in a proper yocto environment make sure linux-libc-headers are properly created from your target kernel and not from native.
> 
> While in general I agree, if the define can be checked, that will include benefits for more users including those with backported kernels.

👍 am building with LE master (which has the updated headers) - was just addressing other builders that might not have the right headers.